app/coffee/BucketController.coffee

```diff
@@ -0,0 +1,36 @@
+PersistorManager = require("./PersistorManager")
+settings = require("settings-sharelatex")
+logger = require("logger-sharelatex")
+FileHandler = require("./FileHandler")
+metrics = require("metrics-sharelatex")
+parseRange = require('range-parser')
+Errors = require('./Errors')
+
+oneDayInSeconds = 60 * 60 * 24
+maxSizeInBytes = 1024 * 1024 * 1024 # 1GB
+
+module.exports = BucketController =
+
+	getFile: (req, res)->
+		{bucket} = req
+		key = req[0]
+		{format, style} = req.query
+		credentials = settings.filestore.s3&[bucket]
```

Comment:
Is the & meant to be there? I've not seen it before. or are you trying todo a null check?

Reply:
I'm trying to do a null check.

Reply:
`settings.filestore.s3?[bucket]` would null check if s3 exists

Reply:
What I want is that if s3 doesn't exist, or if s3[bucket] doesn't exist, you get a null value and don't throw an error.